packages/datetime/src/shortcuts.tsx

```diff
@@ -10,9 +10,23 @@ import { DATERANGEPICKER_SHORTCUTS } from "./common/classes";
 import { clone, DateRange, isDayRangeInRange } from "./common/dateUtils";
 
 export interface IDateRangeShortcut {
+    /** Shortcut label that appears in the list. */
     label: string;
+
+    /**
+     * Date range represented by this shortcut. Note that time components of a
+     * shortcut are ignored by default; set `includeTime: true` to respect them.
+     */
     dateRange: DateRange;
-    shouldChangeTime?: boolean;
```

Comment:
this optional field is the big offender. did not present a compile fail because it is optional so the correct `IDateRangeShortcut` with `includeTime?` is assignable to this incorrect type.